### PR TITLE
fix(coderd/notifications): simplify TemplateWorkspaceManualBuildFailed

### LINKED ISSUE #15065: TemplateWorkspaceManualBuildFailed renders an empty template display name
Re https://github.com/coder/coder/pull/14967#discussion_r1793470910:

> I wondered about this, but didn't want to assume I could overwrite what was specified.
Will update.

@SasSwart @dannykopping  Right now the result looks strange. Please take a look below at the notification I received today, it needs more love. I still don't think we need to refer to the _display name_ specifically.

Screenshot 2024-10-12 at 21 18 40

_Originally posted by @mtojek in https://github.com/coder/coder/issues/14967#issuecomment-2408667169_

### PR DESCRIPTION
**Pull request by SasSwart**
This PR closes #15065.

As advised by @mtojek, a template's display name may be set to "", which is not useful in an email notification. We'd like to provide a friendly name for the template, but it also needs to be identifiable. 

As such, we fall back to template.Name in the case that the template's display name is empty.

### CODE REVIEW

File: coderd/provisionerdserver/provisionerdserver.go [hunk 1]
```diff
@@ -1124,16 +1124,19 @@ func (s *server) notifyWorkspaceManualBuildFailed(ctx context.Context, workspace
...
 	for _, templateAdmin := range templateAdmins {
+		labels := map[string]string{
+			"name":                     workspace.Name,
+			"template_name":            template.DisplayName,
+			"template_version_name":    templateVersion.Name,
+			"initiator":                build.InitiatorByUsername,
+			"workspace_owner_username": workspaceOwner.Username,
+			"workspace_build_number":   strconv.Itoa(int(build.BuildNumber)),
+		}
+		if template.DisplayName == "" {
```
Comment (dannykopping) [inline coderd/provisionerdserver/provisionerdserver.go:1135]: Nit: stylistically I'd prefer do this above the labels definition and assign the value once to either the default or the alternative value.

**Review by dannykopping — APPROVED** [review]
LGTM with nit.

File: coderd/provisionerdserver/provisionerdserver.go (hunk 1 above)
Comment (mtojek) [reply coderd/provisionerdserver/provisionerdserver.go:1135]: +1

**Review by mtojek — COMMENTED** [review]
Fix LGTM 👍 

I believe you can quickly fix it for `TemplateTemplateDeleted` too.

File: coderd/provisionerdserver/provisionerdserver.go (hunk 1 above)
Comment (SasSwart) [reply coderd/provisionerdserver/provisionerdserver.go]: Done

**Pull request merged**